### PR TITLE
doc: some misc doxygen changes to polish I3C and userspace APIs and structs

### PR DESCRIPTION
**Pull request by dcpleung**
(See individual commits for details.)

### CODE REVIEW

File: include/zephyr/drivers/i3c.h [hunk 1]
```diff
@@ -56,19 +56,71 @@ extern "C" {
...
+
+/** @brief Bus Characteristic Register (BCR) - IBI Request Capable bit. */
 #define I3C_BCR_IBI_REQUEST_CAPABLE			BIT(1)
+
+/**
+ * @brief Bus Characteristic Register (BCR) - IBI Payload bit.
+ *
+ * 0 - No data bytes following the accepted IBI.
+ * 1 - One data byte (MDB, Mandatory Data Byte) follows the accepted IBI.
+ *     Additional data byes may also follows.
```
Comment (kartben) [inline include/zephyr/drivers/i3c.h:76]:
```suggestion
 *     Additional data bytes may also follows.
```

File: include/zephyr/drivers/i3c.h [hunk 2]
```diff
@@ -56,19 +56,71 @@ extern "C" {
  *   - 0: No Limitation
  *   - 1: Limitation obtained via GETMXDS CCC.
  */
+
```
Comment (kartben) [inline include/zephyr/drivers/i3c.h:59]: for each group of macros defining the various bits of a register, you may want to consider putting them in `@name { ... }` sections

```c
/**
 * @name Bus Characteristic Register (BCR) fields.
 * @{
 */

...

/** @} */
```

File: include/zephyr/drivers/i3c.h [hunk 3]
```diff
@@ -56,19 +56,71 @@ extern "C" {
...
+/**
+ * @brief Bus Characteristic Register (BCR) - Max Data Speed Limitation bit.
```
Comment (kartben) [inline include/zephyr/drivers/i3c.h:61]: If you go for the suggestion above, you may then want to simplify as the following:

```suggestion
 * @brief Max Data Speed Limitation bit.
```

File: include/zephyr/drivers/i3c.h [hunk 4]
```diff
@@ -56,19 +56,71 @@ extern "C" {
...
+ * 1 - One data byte (MDB, Mandatory Data Byte) follows the accepted IBI.
+ *     Additional data byes may also follows.
+ */
 #define I3C_BCR_IBI_PAYLOAD_HAS_DATA_BYTE		BIT(2)
+
+/**
+ * @brief Bus Characteristic Register (BCR) - Offline Capable bit.
+ *
+ * 0 - Will always response to I3C commands.
+ * 1 - Will not always response to I3C commands.
```
Comment (kartben) [inline include/zephyr/drivers/i3c.h:84]:
```suggestion
 * 0 - Will always respond to I3C commands.
 * 1 - Will not always respond to I3C commands.
```

File: include/zephyr/sys/arch_interface.h [hunk 5]
```diff
@@ -787,10 +787,12 @@ int arch_buffer_validate(void *addr, size_t size, int write);
  * This call returns the optimal virtual address alignment in order to permit
  * such optimization in the following MMU mapping call.
  *
- * @param[in] phys Physical address of region to be mapped, aligned to MMU_PAGE_SIZE
- * @param[in] size Size of region to be mapped, aligned to MMU_PAGE_SIZE
+ * @param[in] phys Physical address of region to be mapped,
+ *                 aligned to CONFIG_MMU_PAGE_SIZE
+ * @param[in] size Size of region to be mapped,
+ *                 aligned to CONFIG_MMU_PAGE_SIZE
```
Comment (kartben) [inline include/zephyr/sys/arch_interface.h:793]:
```suggestion
 * @param[in] phys Physical address of region to be mapped,
 *                 aligned to @kconfig{CONFIG_MMU_PAGE_SIZE}
 * @param[in] size Size of region to be mapped,
 *                 aligned to @kconfig{CONFIG_MMU_PAGE_SIZE}
```

**Review by kartben — CHANGES_REQUESTED** [review]
This is awesome - thanks for taking the time to go through this.

Only spotted a few minor typos. With regards to using `@name`d sections, I will leave it up to you to decide if you want to do it. I think this would help make the likes of `ccc.h` easier to quickly parse when browsing the docs, but it's not like everything is not already nicely grouped alphabetically.

File: include/zephyr/drivers/i3c.h (hunk 1 above)
Comment (dcpleung) [reply include/zephyr/drivers/i3c.h]: Fixed.

File: include/zephyr/drivers/i3c.h (hunk 2 above)
Comment (dcpleung) [reply include/zephyr/drivers/i3c.h:62]: Moved BCR and DCR under named section.

File: include/zephyr/drivers/i3c.h (hunk 3 above)
Comment (dcpleung) [reply include/zephyr/drivers/i3c.h]: Changed.

File: include/zephyr/drivers/i3c.h (hunk 4 above)
Comment (dcpleung) [reply include/zephyr/drivers/i3c.h]: Fixed.

File: include/zephyr/sys/arch_interface.h (hunk 5 above)
Comment (dcpleung) [reply include/zephyr/sys/arch_interface.h]: Does it generate link to the kconfig doc? My local build doesn't link anything but simply making it with monospace fonts.

Comment (kartben) [reply include/zephyr/sys/arch_interface.h]: Yes but only when rendered on the main docs website, not in the Doxygen docs itself

https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/zephyr.doxyfile.in#L278

**Review by kartben — APPROVED** [review]
Very nice, thanks! 👏🏼

**Review by peter-mitsis — APPROVED** [review]

**Review by XenuIsWatching — APPROVED** [review]

**Pull request merged**